checkstyle-tester/diff.groovy

```diff
@@ -0,0 +1,232 @@
+import java.nio.file.Paths
+
+import static java.lang.System.err
+
+def cli = new CliBuilder(usage:'groovy diff.groovy [options]')
+cli.with {
+    r(longOpt: 'localGitRepo', args: 1, argName: 'localGitRepo', 'Path to local git repository')
+    b(longOpt: 'patchBranch', args: 1, argName: 'patchBranch', 'Name of the patch branch in local git repository')
+    c(longOpt: 'checkstyleCfg', args: 1, argName: 'checkstyleCfg', 'Path to checkstyle config file')
+    p(longOpt: 'projectsToTestOn', args: 1, argName: 'projectsToTestOn', 'Path to file which contains projects to test on')
+}
```

Comment:
we need to have:
path to repo - "r"
base branch name ("master" is default) - "b"
patched/changed branch name - "p"
checktyle config - "c"
path to properties file with projects - "l" (list of projects)